src/core/border_router/routing_manager.cpp

```diff
@@ -587,18 +583,11 @@ void RoutingManager::EvaluateOnLinkPrefix(void)
 
         mOnLinkPrefixDeprecateTimer.Stop();
     }
-    // When an application-specific on-link prefix is received and it is bigger than the
-    // advertised prefix, we will not remove the advertised prefix. In this case, there
-    // will be two on-link prefixes on the infra link. But all BRs will still converge to
-    // the same smallest on-link prefix and the application-specific prefix is not used.
     else if (mIsAdvertisingLocalOnLinkPrefix)
     {
-        if (!(mLocalOnLinkPrefix < *smallestOnLinkPrefix))
-        {
-            LogInfo("EvaluateOnLinkPrefix: There is already smaller on-link prefix %s on %s",
-                    smallestOnLinkPrefix->ToString().AsCString(), mInfraIf.ToString().AsCString());
-            DeprecateOnLinkPrefix();
-        }
+        LogInfo("EvaluateOnLinkPrefix: There is an on-link prefix %s on %s",
```

Comment:
Quick note here to help with review:
I removed the logic of comparing against smallest prefix and instead we just check if we have already discovered an acceptable on-link prefix, and we are  `mIsAdvertisingLocalOnLinkPrefix` we `DeprecateOnLinkPrefix()`.